api/src/main/java/org/pmiops/workbench/api/OfflineTestUsersController.java

```diff
@@ -80,57 +84,59 @@ private void ensureTosCompliance(String username) {
 
   @Override
   public ResponseEntity<Void> deleteAllTestUserWorkspaces() {
-    WorkbenchConfig config = workbenchConfigProvider.get();
-    WorkbenchConfig.E2ETestUserConfig testUserConf = config.e2eTestUsers;
+    LOGGER.info("Deleting test user workspaces...");
 
-    // only some environments have test users
-    if (testUserConf == null) {
-      LOGGER.info("This environment does not have a test user config block.  Exiting.");
-    } else {
-      taskQueueService.groupAndPushDeleteTestWorkspaceTasks(
-          testUserConf.testUserEmails.stream()
-              .flatMap(this::enumerateAoUWorkspaces)
-              .collect(Collectors.toList()));
-    }
+    deleteAllWorkspaces(
+        this::enumerateAoUWorkspaces, taskQueueService::groupAndPushDeleteTestWorkspaceTasks);
 
+    LOGGER.info("Done deleting test user workspaces.");
     return ResponseEntity.ok().build();
   }
 
   @Override
   public ResponseEntity<Void> deleteAllTestUserWorkspacesOrphanedInRawls() {
+    LOGGER.info("Deleting test user workspaces in Rawls (Terra)...");
+
+    deleteAllWorkspaces(
+        this::enumerateRawlsWorkspaces,
+        taskQueueService::groupAndPushDeleteTestWorkspaceInRawlsTasks);
+
+    LOGGER.info("Done deleting test user workspaces in Rawls (Terra)...");
+    return ResponseEntity.ok().build();
+  }
+
+  private <T> void deleteAllWorkspaces(
+      Function<String, Stream<T>> enumerator, Consumer<List<T>> queue) {
     WorkbenchConfig config = workbenchConfigProvider.get();
     WorkbenchConfig.E2ETestUserConfig testUserConf = config.e2eTestUsers;
 
     // only some environments have test users
     if (testUserConf == null) {
       LOGGER.info("This environment does not have a test user config block.  Exiting.");
     } else {
-      taskQueueService.groupAndPushDeleteTestWorkspaceInRawlsTasks(
-          testUserConf.testUserEmails.stream()
-              .flatMap(this::enumerateRawlsWorkspaces)
-              .collect(Collectors.toList()));
+      queue.accept(testUserConf.testUserEmails.stream().flatMap(enumerator).toList());
     }
-
-    return ResponseEntity.ok().build();
   }
 
   private Stream<TestUserWorkspace> enumerateAoUWorkspaces(String username) {
     List<WorkspaceResponse> workspaces = impersonatedWorkspaceService.getOwnedWorkspaces(username);
+    String action = workspaces.isEmpty() ? "." : "; queueing for deletion";
```

Comment:
most of the time this logged "user has 0 workspaces; queuing..." which was not accurate because it doesn't queue when there are 0